### PR TITLE
clean up duplicate names for conformance tests

### PR DESCRIPTION
**Pull request by xd1313113**
Issue #, if available:
duplicate names in conformance tests will result in conformance report inaccurate and confuse test owners.

Also, long test name get truncated in Junit test framework in the testcallbacks and also leading to duplicate test names
Description of changes:
Rename duplicate tests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by xd1313113** [comment]
> Shall we add a validating test in runner to detect duplicate names to prevent the same issue in the future?

Duplicates detection is added in this PR https://github.com/partiql/partiql-lang-kotlin/pull/1833

**Comment by xd1313113** [comment]
> Shall we add a validating test in runner to detect duplicate names to prevent the same issue in the future?

Yes, Duplicates detection is added in this PR https://github.com/partiql/partiql-lang-kotlin/pull/1833

File: partiql-tests-data/eval/ion/query/group-by/group-by.ion [hunk 1]
```diff
@@ -49,7 +49,7 @@
     ],
   },
   {
-    name:"group by with where",
+    name:"group by with where fiscal_year >= DATE",
```
Comment (XuechunHHH) [inline partiql-tests-data/eval/ion/query/group-by/group-by.ion:52]: I think it would be better to also use `ion timestamp` for conformance.

Comment (xd1313113) [reply partiql-tests-data/eval/ion/query/group-by/group-by.ion:52]: there are two versions of this test, One is Date literal and other one is ion timestamp. rename to avoid name conflicts.

File: partiql-tests-data/eval/primitives/functions/mod.ion [hunk 2]
```diff
@@ -71,18 +71,6 @@ mod::[
       output:1
     }
   },
-  {
```
[the remark this answers is not in the document]
Comment (xd1313113) [reply partiql-tests-data/eval/primitives/functions/mod.ion:74]: Just same as https://github.com/partiql/partiql-tests/blob/4c50e2c0f7a07bd75830c6cd616b53a7cb01ac8b/partiql-tests-data/eval/primitives/functions/mod.ion#L27

File: partiql-tests-data/partiql-tests-schema.isl [hunk 3]
```diff
@@ -37,7 +37,8 @@ type::{
     name: TestCase,
     type: struct,
     fields: {
-        name: { type: string, occurs: required },
+        // Junit test framework callbacks have 512 character limit on display name. Set to 350 to reverse some for system
```
[the remark this answers is not in the document]
Comment (xd1313113) [reply partiql-tests-data/partiql-tests-schema.isl:40]: Yeah, we need to reserve some characters for runner appended text. I pick (512-350) = 162 for system use. For now we are appending `compileOption: PERMISSIVE`

File: partiql-tests-data/eval/primitives/functions/char_length.ion [hunk 4]
```diff
@@ -1068,7 +1068,8 @@ char_length::[
...
+    // name:"char_length valid cases{in:\"\\u0401\\u0402\\u0403\\u0404\\u0405\\u0406\\u0407\\u0408\\u0409\\u040a\\u040b\\u040c\\u040d\\u040e\\u040f\\u0410\\u0411\\u0412\\u0413\\u0414\\u0415\\u0416\\u0417\\u0418\\u0419\\u041a\\u041b\\u041c\\u041d\\u041e\\u041f\\u0420\\u0421\\u0422\\u0423\\u0424\\u0425\\u0426\\u0427\\u0428\\u0429\\u042a\\u042b\\u042c\\u042d\\u042e\\u042f\\u0430\\u0431\\u0432\\u0433\\u0434\\u0435\\u0436\\u0437\\u0438\\u0439\\u043a\\u043b\\u043c\\u043d\\u043e\\u043f\\u0440\\u0441\\u0442\\u0443\\u0444\\u0445\\u0446\\u0447\\u0448\\u0449\\u044a\\u044b\\u044c\\u044d\\u044e\\u044f\",result:(success 79)}"
+    name:"char_length valid cases{Complete Cyrillic Alphabet}",
```
[the remark this answers is not in the document]
Comment (xd1313113) [reply partiql-tests-data/eval/primitives/functions/char_length.ion:1072]: Let us cut an issue as there are a lot of them. Want to get this PR in first. https://github.com/partiql/partiql-tests/issues/163

**Review by alancai98 — APPROVED** [review]
